base/src/main/java/io/spine/type/KnownTypes.java

```diff
@@ -20,61 +20,65 @@
 
 package io.spine.type;
 
+import com.google.errorprone.annotations.Immutable;
 import com.google.protobuf.Any;
 import com.google.protobuf.Message;
 import com.google.protobuf.util.JsonFormat;
 import io.spine.annotation.Internal;
 import io.spine.code.proto.FileSet;
 import io.spine.code.proto.Type;
 import io.spine.code.proto.TypeSet;
+import io.spine.security.InvocationGuard;
 
+import java.io.Serializable;
 import java.util.Optional;
 import java.util.Set;
-import java.util.function.Supplier;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
-import static com.google.common.base.Suppliers.memoize;
+import static com.google.common.base.Preconditions.checkNotNull;
 import static java.util.stream.Collectors.toSet;
 
 /**
  * All Protobuf types known to the application.
  *
  * <p>This includes types generated by Protobuf compiler and collected into
  * the {@linkplain io.spine.code.proto.FileDescriptors#KNOWN_TYPES resource file}.
- *
- * @author Mikhail Mikhaylov
- * @author Alexander Litus
- * @author Alexander Yevsyukov
- * @author Dmytro Dashenkov
  */
 @Internal
-public enum KnownTypes {
+@Immutable
```

Comment:
Please document the limitations of immutability.

Reply:
Done.